app/javascript/packs/foremCreatorSignup.js

```diff
@@ -0,0 +1,98 @@
+function setDefaultUsername(event) {
+  if (
+    document
+      .getElementsByClassName('js-creator-signup-username-row')[0]
+      .classList.contains('hidden')
+  ) {
+    const name = event.target.value;
+    // It's the first user and so we can assume that this username is not taken.
+    const usernameHint = createUsernameHint(name);
+    setUsernameHint(usernameHint);
+    setUsernameField(usernameHint);
+    showHintRow();
+  }
+}
+
+function createUsernameHint(name) {
+  return name
+    .toLowerCase()
+    .replace(/[^a-zA-Z0-9]/g, '_')
+    .substr(0, 30);
+}
+
+function showHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.remove('hidden');
+}
+
+function setUsernameHint(usernameHint) {
+  const usernameHintDisplay = document.getElementsByClassName(
+    'js-creator-signup-username-hint',
+  )[0];
+  usernameHintDisplay.innerHTML = usernameHint;
+}
+
+function setUsernameField(usernameHint) {
+  const usernameField = document.getElementsByClassName(
+    'js-creator-signup-username',
+  )[0];
+  usernameField.value = usernameHint;
+}
+
+function showUsernameField() {
+  const usernameRow = document.getElementsByClassName(
+    'js-creator-signup-username-row',
+  )[0];
+  usernameRow.classList.remove('hidden');
+  focusUsernameInput(usernameRow);
+  hideHintRow();
+}
+
+function focusUsernameInput(usernameRow) {
+  // A timer with a count of 0 will run when the thread becomes idle
+  window.setTimeout(() => {
+    usernameRow.getElementsByTagName('input')[0].focus();
+  }, 0);
+}
+
+function hideHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
+  hintRow.classList.add('hidden');
+}
+
+function togglePasswordMask() {
+  visible = !visible;
+  togglePasswordType(visible);
+  toggleEyeIcons(visible);
+}
+
+function togglePasswordType(visible) {
+  const passwordType = visible ? 'text' : 'password';
+  passwordField.type = passwordType;
+}
+
+function toggleEyeIcons(visible) {
+  eyeOffIcon.classList.toggle('hidden', !visible);
+  eyeIcon.classList.toggle('hidden', visible);
+}
+
+let visible = false;
+const eyeIcon = document.getElementsByClassName('js-eye')[0];
+const eyeOffIcon = document.getElementsByClassName('js-eye-off')[0];
```

Comment:
Refactor (blocking): Instead of two elements, opt for toggling `js-eye-off`.

e.g. Instead of

```erb
          <span class="js-creator-password-visibility">
            <%= inline_svg_tag("eye.svg", class: "absolute bottom-2 right-2 cursor-pointer js-eye", aria: true, title: "Eye icon") %>
            <%= inline_svg_tag("eye-off.svg", class: "absolute bottom-2 right-2 cursor-pointer hidden js-eye-off", aria: true, title: "Eye Off icon") %>
          </span>
```

opt for

```erb
          <span class="js-creator-password-visibility">
            <%= inline_svg_tag("eye.svg", class: "absolute bottom-2 right-2 cursor-pointer js-eye", aria: true, title: "Eye icon") %>
          </span>
```

As well, since this is something the user can click, it should be wrapped in a button and be made a [toggle button](https://dequeuniversity.com/library/aria/button-toggle). We have an example in the repository that was merged recently. See #14256 cc: @aitchiss

Reply:
Nice spot - it's a great place for a toggle button! I couldn't get the current implementation to work at all in Chrome, but I'm sure making this a button will fix the issue.

Another thing I noticed - throughout the SVGs used in this PR we have `title` attributes that describe the icon name - the `title` is visible in the UI if you hover long enough, so I think if we want a title it should be something like "Toggle password visibility" so it makes sense to the user (there's a few others, e.g. the Forem background too)

Once this is a button I'd also change `aria: true` to `aria_hidden: true` and give the button an aria-label instead. Assistive tech is very inconsistent when we rely on SVG titles for labels, so it's just a bit more robust 🙂

Reply:
Thanks for these tips @nickytonline and @aitchiss - makes sense. I'll amend accordingly. ✨

Reply:
e.g. Instead of
```
          <span class="js-creator-password-visibility">
            <%= inline_svg_tag("eye.svg", class: "absolute bottom-2 right-2 cursor-pointer js-eye", aria: true, title: "Eye icon") %>
            <%= inline_svg_tag("eye-off.svg", class: "absolute bottom-2 right-2 cursor-pointer hidden js-eye-off", aria: true, title: "Eye Off icon") %>
          </span>
```
opt for
```
          <span class="js-creator-password-visibility">
            <%= inline_svg_tag("eye.svg", class: "absolute bottom-2 right-2 cursor-pointer js-eye", aria: true, title: "Eye icon") %>
          </span>
```

@nickytonline not sure I understand this feedback. 
Just having the one icon, the `eye.svg`, wouldn't be sufficient because we show the eye svg when the password is in a masked state to indicate the unmask action and then the eye-off svg (which has the / over the eye) when the password is not masked to indicate the mask action.

Reply:
What I meant was to toggle the SVG based on the CSS class, but now that I think of it, as long as this is wrapped in a button, you change the SVG accordingly. Otherwise, you can't leverage the `inline_svg_tag` from the asset pipeline.

Reply:
Done!